MediaBrowser.Providers/Plugins/Tmdb/TmdbUtils.cs

```diff
@@ -188,5 +192,66 @@ public static string BuildParentalRating(string countryCode, string ratingValue)
 
             return newRating.Replace("DE-", "FSK-", StringComparison.OrdinalIgnoreCase);
         }
+
+        /// <summary>
+        /// Removes invalid characters for file names from the search result name for better comparsison.
+        /// </summary>
+        /// <param name="searchResultName">The name from the search result.</param>
+        /// <returns>The input with invalid file name characters removed.</returns>
+        public static string? RemoveInvalidFileCharacters(string? searchResultName)
+        {
+            if (searchResultName == null)
+            {
+                return null;
+            }
+
+            string formatted = _fileSystemInvalid.Replace(searchResultName, string.Empty);
+            return formatted;
+        }
+
+        /// <summary>
+        /// Identifies the nearest match to the search criteria. Addresses issues with search results from TMDB.
+        /// </summary>
+        /// <param name="name">The cleaned up name searched for.</param>
+        /// <param name="year">The year searched for.</param>
+        /// <param name="searchResults">The search results from a Movie or Tv search.</param>
+        /// <returns>The nearest match to the search criteria.</returns>
+        public static SearchMovieTvBase? GetBestMatch(string name, int year, IEnumerable<SearchMovieTvBase> searchResults)
+        {
+            var filtered = searchResults;
+
+            // partial matches can be returned before exact matches. Look for exact matches after removing invalid characters
+            var nearExactMatchFiltered = filtered.Where(s =>
+            {
+                var resultName = (s as SearchMovie)?.Title ?? (s as SearchTv)?.Name;
+                var resultOriginalName = (s as SearchMovie)?.OriginalTitle ?? (s as SearchTv)?.OriginalName;
+
+                return RemoveInvalidFileCharacters(resultName) == name
+                    || RemoveInvalidFileCharacters(resultOriginalName) == name;
```

Comment:
```suggestion
                string? resultName;
                string? resultOriginalName;
                switch (s)
                {
                    case SearchMovie searchMovie:
                        resultName = searchMovie.Title;
                        resultOriginalName = searchMovie.OriginalTitle;
                        break;
                    case SearchTv searchTv:
                        resultName = searchTv.Name;
                        resultOriginalName = searchTv.OriginalName;
                        break;
                    default:
                        return false;
                }

                return string.Equals(RemoveInvalidFileCharacters(resultName), name, StringComparison.OrdinalIgnoreCase)
                       || string.Equals(RemoveInvalidFileCharacters(resultOriginalName), name, StringComparison.OrdinalIgnoreCase);
```

Longer code but more maintainable